trendmicrovisionone_connector.py

```diff
@@ -835,850 +733,1438 @@ def _handle_remove_from_blocklist(self, param):
+
+        if "error" in response.result_code.lower():
             self.debug_print(
-                "REST call failed, please check your endpoints and/or params"
+                "Something went wrong, please check workbench_id and content."
+            )
+            raise RuntimeError(
+                f"Error adding note to workbench {workbench_id}. Result Code: {response.error}"
             )
-            return action_result.get_status()
 
+        assert response.response is not None
+        location = response.response.location
+        note_id = location.split("/")[-1]
+        msg = "success"
+        result = {"note_id": note_id, "message": msg}
         # Add the response into the data section
-        action_result.add_data(response)
+        action_result.add_data(result)
 
         # Return success
         return action_result.set_status(phantom.APP_SUCCESS)
 
-    def _handle_delete_email_message(self, param):
-        # send progress messages back to the platform
+    def _handle_update_status(self, param):
+        """
+        Modifies the status of an alert or investigation triggered in Workbench.
+        Args:
+            workbench_id(str): Workbench alert ID (e.g. WB-14-20190709-00003)
+            status(str): Status to be updated ("New" "In Progress" "True Positive" "False Positive" "Benign True Positive" "Closed").
+            if_match(str): Target resource will be updated only if it matches ETag of the target one (e.g. "d41d8cd98f00b204e9800998ecf8427e").
+        Returns:
+            message(str): Success or Failure.
+        """
         self.save_progress(
             "In action handler for: {0}".format(self.get_action_identifier())
         )
 
         # Add an action result object to self (BaseConnector) to represent the action for this param
         action_result = self.add_action_result(ActionResult(dict(param)))
 
-        # Param setup
-        message_id = param["message_id"]
-        mailbox = param["mailbox"]
-        message_delivery_time = param.get("message_delivery_time", "")
-        product_id = param.get("product_id", "sao")
-        description = param.get("description", "Delete e-mail.")
-        body = {
-            "messageId": message_id,
-            "mailBox": mailbox,
-            "messageDeliveryTime": message_delivery_time,
-            "productId": product_id,
-            "description": description,
-        }
+        # Required Params
+        workbench_id = param["workbench_id"]
+        status = param["status"]
+        if_match = param["if_match"]
+
+        # Initialize Pytmv1
+        client = self._get_client()
+
+        # Choose Status Enum
+        sts = status.upper()
+        if sts in InvestigationStatus.__members__:
+            status = InvestigationStatus[sts]
+        else:
+            self.debug_print("Something went wrong, please check input params.")
+            raise RuntimeError(f"Please check status: {status}")
```

Comment:
I think an inverted condition here would be better for consistency with the rest of the code

```python
if sts not in InvestigationStatus.__members__:
            self.debug_print("Something went wrong, please check input params.")
            raise RuntimeError(f"Please check status: {status}")

status = InvestigationStatus[sts]
```